### PR TITLE
Add some useful additional printer columns for TidbMonitor 

### PR DESCRIPTION
**Pull request by mikechengwei**
### What problem does this PR solve? 
#2169 

### What is changed and how does it work?
1.  add `ready`, `updatedReplicas` and `availableReplicas` columns  to tidbmonitor status and sync .
2. kubectl support printing `READY`  ,`UP-TO-DATE`  and AVAILABLE columns 

We use  deployment to deploy tidbmonitor , so we can print same column as the deployment.
### Check List 

 - Unit test
 - E2E test
 - Stability test
 - Manual test (add detailed scripts or steps below)

Code changes

 - Has Go code change

Side effects

 - None

### CODE REVIEW

**Comment by mikechengwei** [comment]
@cofyc Can you help me review it?

File: pkg/apis/pingcap/v1alpha1/tidbmonitor_types.go [hunk 1]
```diff
@@ -173,8 +173,17 @@ type TidbClusterRef struct {
 	Name string `json:"name"`
 }
 
-// TODO: sync status
+// TidbMonitorStatus is the current status of
```
Comment (cofyc) [inline pkg/apis/pingcap/v1alpha1/tidbmonitor_types.go:176]:
```suggestion
// TidbMonitorStatus represents information about the status of a TidbMonitor.
```

File: pkg/apis/pingcap/v1alpha1/tidbmonitor_types.go [hunk 2]
```diff
@@ -173,8 +173,17 @@ type TidbClusterRef struct {
...
 type TidbMonitorStatus struct {
+
+	// Ready is the number of ready containers
+	Ready corev1.ConditionStatus `json:"ready,omitempty"`
+
+	// Total number of non-terminated pods targeted by this deployment that have the desired template spec.
+	UpdatedReplicas int32 `json:"updatedReplicas,omitempty"`
+
+	// Total number of available pods (ready for at least minReadySeconds) targeted by this tidbMonitor.
+	AvailableReplicas int32 `json:"availableReplicas,omitempty"`
```
Comment (cofyc) [inline pkg/apis/pingcap/v1alpha1/tidbmonitor_types.go:186]: maybe it's better to just add a  `DeploymentStatus` field to hold the underlying deployment status

File: pkg/util/crdutil.go [hunk 3]
```diff
@@ -218,6 +218,25 @@ var (
 		Type:     "date",
 		JSONPath: ".metadata.creationTimestamp",
 	}
+
+	tidbMonitorAdditionalPrinterColumns      []extensionsobj.CustomResourceColumnDefinition
+	readyTiDBMonitorAdditionalPrinterColumns = extensionsobj.CustomResourceColumnDefinition{
+		Name:     "READY",
+		Type:     "string",
+		JSONPath: ".status.ready",
```
Comment (cofyc) [inline pkg/util/crdutil.go:226]:
```suggestion
		JSONPath: `.status.deployment.conditions[?(@.type=="Available")].status`,
```

can we think the TidbMonitor is ready when the underlying deployment's `Available` condition is true?

**Comment by cofyc** [comment]
> [CLA assistant check](https://cla-assistant.io/pingcap/tidb-operator?pullRequest=2967)
> Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/pingcap/tidb-operator?pullRequest=2967) before we can accept your contribution.
> You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/pingcap/tidb-operator?pullRequest=2967) it.

It seems that you use new email in the commits, can you sign the CLA check again?

**Comment by mikechengwei** [comment]
reopen this pr

**Comment by cofyc** [comment]
I found that we didn't follow the same name convention of Statefulset on PVC objects. Do you think is it possible to provide a smooth migration without manual intervention?

If not, maybe we can have a field `controller` field to specify which workload we're going to use? It has two fields: Deployment, StatefulSet. New users can use `StatefulSet`.

**Comment by cofyc** [comment]
Sorry, I posted on the wrong issue.

**Pull request closed without merging**